torch/_dynamo/variables/torch_function.py

```diff
@@ -124,24 +124,34 @@ def get_mode_index(cls, ind):
 
 
 class TorchFunctionModeVariable(ContextWrappingVariable):
-    def __init__(self, value, **kwargs):
+    def __init__(self, value, source=None, **kwargs):
         super().__init__(value, **kwargs)
+        assert source
         self.value = value
-
-    @staticmethod
-    def get_global_mangled_name(tx, val):
-        return get_safe_global_name(
-            tx, f"__torch_function_mode_{val.__class__.__name__}", val
-        )
+        self.source = source
 
     def reconstruct(self, codegen):
         # We don't support locally created torch function modes yet
         assert self.source
         self.source.reconstruct(codegen)
 
+    def python_type(self):
+        return type(self.value)
+
     def _call_func(self, tx, values):
         unimplemented("torch function mode context manager is not supported yet")
 
+    def call_torch_function(self, tx: "InstructionTranslator", fn, types, args, kwargs):
```

Comment:
What was the state of TorchFunctionMode before this PR? I see a TorchFunctionMode variable already

Reply:
I landed these PRs as part of the stack previously (once landed they got removed from this stack's description)

https://github.com/pytorch/pytorch/pull/133132